File: ecm-core/src/main/java/br/com/samuel/app/resources/roles/CountRolesResource.java
package br.com.samuel.app.resources.roles;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import br.com.samuel.app.resources.interfaces.ICounterResource;
import br.com.samuel.app.usecases.models.CountElements;
import br.com.samuel.app.usecases.roles.CountRoles;

@RestController
@RequestMapping("/roles")
public class CountRolesResource extends ICounterResource<CountRoles> {

    @GetMapping("/count")
    public ResponseEntity<CountElements> run() { 
        return ResponseEntity.ok(counter().run());
    }
}

File: ecm-view/src/app/resources/roles/count-roles.resource.ts
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";

import { ICounterResource } from "../interfaces/counter.resource";
import { CountElements } from "src/app/utils/models/count-elements.entity";

@Injectable({
    providedIn: 'root'
})
export class CountRolesResource extends ICounterResource {

    constructor() {
        super('roles', 'count')
    }

    run(): Observable<CountElements> {
        return this.http.get<CountElements>(this.baseUrl());
    }
}